dashboard/test/helpers/levels_helper_test.rb

```diff
@@ -267,4 +267,22 @@ def stub_country(code)
     assert_equal false, app_options[:level]['submittable']
   end
 
+  test 'Milestone posting is disabled if Gatekeeper flag is set' do
+    @level = create(:level, :blockly)
+    @script = create :script
+
+    Gatekeeper.set('postMilestone', where: {script_name: @script.name}, value: false)
+    app_options = self.app_options
+    assert_equal false, app_options[:postMilestone]
+  end
+
+  test 'Milestone posting is enabled for other levels' do
+    @level = create(:level, :blockly)
+    @script = create :script
+
+    Gatekeeper.set('postMilestone', where: {script_name: 'Not the level name'}, value: false)
```

Comment:
Not before November 9th, but probably before HOC, we will want an integration test to verify that the milestone posting is really disabled.  This will be tricky to test given that it cuts across the client and the server.